### PR TITLE
common-cpu-intel-alder-lake: init

### PR DESCRIPTION
**Pull request by andresilva**
###### Description of changes

Creates a new config for intel alder lake (using the existing gpu module). Also updates the thinkpad x1 6th gen config to import the cpu architecture-specific module.

###### Things done

- [ ] Tested the changes in your own NixOS Configuration
- [ ] Tested the changes end-to-end by using your fork of `nixos-hardware` and
      importing it via `<nixos-hardware>` or Flake input

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -301,6 +301,7 @@
         common-cpu-amd-zenpower = import ./common/cpu/amd/zenpower.nix;
         common-cpu-amd-raphael-igpu = import ./common/cpu/amd/raphael/igpu.nix;
         common-cpu-intel = import ./common/cpu/intel;
+        common-cpu-intel-alder-lake = import ./common/cpu/intel/alder-lake;
```
Comment (Mic92) [inline flake.nix:304]: We are actually deprecating those: https://github.com/NixOS/nixos-hardware/issues/992
You can still use them if you want but we might remove them again.

Comment (Mic92) [reply flake.nix:304]:
```suggestion
```

Comment (andresilva) [reply flake.nix:304]: I'm using nixos-hardware as a flake, how can I use these if they are not exported?

Comment (Mic92) [reply flake.nix:304]:
```
{
  imports = [  "${nixos-hardware}/common/cpu/intel/alder-lake" ];
}
```

Comment (Mic92) [reply flake.nix:304]: Just be aware we will remove this after it's in nixpkgs.

Comment (andresilva) [reply flake.nix]: Thanks! 👍

**Comment by Mic92** [comment]
@mergify queue

**Pull request merged**